hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml

```diff
@@ -226,6 +226,24 @@
         - 'kubernetes-e2e-{suffix}'
 
 - project:
+    name: kubernetes-e2e-gce-1-2
+    trigger-job: 'kubernetes-build-1.2'
+    test-owner: 'Build Cop'
+    provider-env: |
+        {gce-provider-env}
+        export JENKINS_PUBLISHED_VERSION="ci/latest-1.2"
```

Comment:
putting this in 'provider-env' seems somewhat arbitrary, but I guess there's no better place?

seems like we may want to have this version as another option for projects? can default to 'ci/latest', but override to 'ci/latest-1.2' here, 'ci/latest-1.1' for 1.1 builds, etc?